src/module.c

```diff
@@ -621,7 +628,7 @@ static int AlterIndexInternalCommand(RedisModuleCtx *ctx, RedisModuleString **ar
       return RedisModule_ReplyWithSimpleString(ctx, "OK");
     }
   }
-  IndexSpec_AddFields(sp, ctx, &ac, initialScan, &status);
+  IndexSpec_AddFields(ref, sp, ctx, &ac, initialScan, &status);
```

Comment:
Why does it need to get the ref and not the spec?

Reply:
because this function might start a BG scanning. we can alternatively search a second time in the dictionary